mobile/src/main/java/ch/epfl/sdp/mobile/ui/puzzles/Puzzle.kt

```diff
@@ -0,0 +1,54 @@
+package ch.epfl.sdp.mobile.ui.puzzles
+
+import androidx.compose.runtime.Composable
+import ch.epfl.sdp.mobile.application.chess.engine.Color.*
+import ch.epfl.sdp.mobile.application.chess.engine.implementation.buildBoard
+import ch.epfl.sdp.mobile.application.chess.engine.rules.Action
+import ch.epfl.sdp.mobile.application.chess.notation.FenNotation
+import ch.epfl.sdp.mobile.application.chess.notation.FenNotation.BoardSnapshot
+import ch.epfl.sdp.mobile.ui.game.ChessBoardState
+import ch.epfl.sdp.mobile.ui.game.ChessBoardState.Color
+
+interface Puzzle {
+  val uid: String
```

Comment:
Does the ui need to know the `Puzzle` id ? You should instead expect a generic `P : Puzzle` in your `PuzzleScreenState`, like we have for generic matches.

Reply:
Fixed in [7d121bf](https://github.com/epfl-SDP/android/pull/300/commits/7d121bf381204e2b65a135371103e840079c372b)